packages/web3torrent/src/clients/web3t-channel-client.ts

```diff
@@ -0,0 +1,193 @@
+import {ChannelResult, Message, ChannelClient} from '@statechannels/channel-client';
+import {bigNumberify} from 'ethers/utils';
+
+interface ChannelState {
+  channelId: string;
+  turnNum: string;
+  challengeExpirationTime;
+  aUserId: string;
+  bUserId: string;
+  aAddress: string;
+  bAddress: string;
+  aOutcomeAddress: string;
+  bOutcomeAddress: string;
+  aBal: string;
+  bBal: string;
+}
+
+// This class wraps the channel client converting the request/response formats to those used in the app
+
+export class Web3TorrentChannelClient {
+  constructor(private readonly channelClient: ChannelClient) {}
+
+  mySigningAddress?: string;
+  myEthereumSelectedAddress?: string; // this state can be inspected to infer whether we need to get the user to "Connect With MetaMask" or not.
```

Comment:
If we don't have `myEthereumSelectedAddress`, we need to display some different UI and then call the getter on this class, which may cause metamask to pop up. Open for discussion about whether we want to store the information about whether this is necessary, here or elsewhere. In RPS (for example) we have an explicit state machine for the app which will not progress until we have these addresses.